### PR TITLE
Add event listener for resize events

### PR DESCRIPTION
**Pull request by ibdafna**
This PR adds an event listener so that `lumino` based widgets rendered in a classic notebook can receive a resize message when a window resize event triggers.

### CODE REVIEW

**Comment by jasongrout** [comment]
Fixes https://github.com/jupyter-widgets/ipywidgets/issues/2467

File: packages/base/src/widget.ts [hunk 1]
```diff
@@ -895,6 +895,18 @@ export class DOMWidgetView extends WidgetView {
           oldLayoutView.remove();
         }
 
+        const resizeFunc = () => {
```
Comment (jtpio) [inline packages/base/src/widget.ts:898]:
```suggestion
        const resizeFunc = (): void => {
```

**Comment by jasongrout** [comment]
@ibdafna, thanks for looking into this.

I think we should register the handler in the classic notebook widget manager *once*, then when it is triggered, have the classic notebook manager go through each of the top-level widgets and trigger the lumino event. And probably do the same for the html manager.

The way things are set up right now in this PR, it looks like every widget will register a handler with window (whether it needs to or not).

@ibdafna - I'm happy to connect over video to walk through this if you want.

**Comment by ibdafna** [comment]
> 
> 
> @ibdafna, thanks for looking into this.
> 
> I think we should register the handler in the classic notebook widget manager _once_, then when it is triggered, have the classic notebook manager go through each of the top-level widgets and trigger the lumino event. And probably do the same for the html manager.
> 
> The way things are set up right now in this PR, it looks like every widget will register a handler with window (whether it needs to or not).
> 
> @ibdafna - I'm happy to connect over video to walk through this if you want.

Thank you - yes, I'll reach out!

File: packages/html-manager/src/htmlmanager.ts [hunk 2]
```diff
@@ -32,6 +34,16 @@ export class HTMLManager extends ManagerBase {
       },
       0
     );
+
+    this._viewList = new Set<DOMWidgetView>();
+    window.addEventListener('resize', () => {
+      this._viewList.forEach(view => {
+        MessageLoop.postMessage(
+          view.lmWidget,
+          Widget.ResizeMessage.UnknownSize
```
Comment (jasongrout) [inline packages/html-manager/src/htmlmanager.ts:43]: Likely this will also need `LuminoWidget.Widget` instead of just `Widget` like in the code below

**Review by jasongrout — COMMENTED** [review]
Thanks! I found a few things while testing, noted below.

**Comment by jasongrout** [comment]
FYI, I rebased on master

**Comment by jasongrout** [comment]
@ibdafna - I rebased and fixed a few issues with the examples, and checked manually that the resize event is now happening for the classic notebook and the html manager. Thanks again!

**Review by jasongrout — APPROVED** [review]

**Pull request merged**